cpp/arcticdb/version/local_versioned_engine.cpp

```diff
@@ -336,28 +336,47 @@ std::pair<VersionedItem, FrameAndDescriptor> LocalVersionedEngine::read_datafram
     return std::make_pair(version.value_or(VersionedItem{}), std::move(frame_and_descriptor));
 }
 
+folly::Future<std::pair<VersionedItem, std::optional<google::protobuf::Any>>> LocalVersionedEngine::get_descriptor(
+    std::optional<AtomKey>&& key,
+    const StreamId& stream_id,
+    const VersionQuery& version_query){
+    missing_data::check<ErrorCode::E_NO_SUCH_VERSION>(static_cast<bool>(key.has_value()),
+        "Unable to retrieve descriptor data. {}@{}: version not found", stream_id, version_query);
+    return store()->read_metadata(key.value())
+    .thenValue([](auto&& key_meta_pair){
+        auto [key_seg, meta] = std::move(key_meta_pair);
+        VersionedItem version{std::move(to_atom(*key_seg))};
+        return std::pair{std::move(version), std::move(meta)};
+    });
+}
+
+folly::Future<std::pair<VersionedItem, std::optional<google::protobuf::Any>>> LocalVersionedEngine::get_descriptor_async(
+    folly::Future<std::optional<AtomKey>>&& version_fut,
+    const StreamId& stream_id,
+    const VersionQuery& version_query){
+    return  std::move(version_fut)
+    .thenValue([this, &stream_id, &version_query](std::optional<AtomKey>&& key){
+        return get_descriptor(std::move(key), stream_id, version_query);
+    });
+}
+
 std::pair<VersionedItem, std::optional<google::protobuf::Any>> LocalVersionedEngine::read_descriptor_internal(
-        const StreamId& stream_id,
-        const VersionQuery& version_query
+    const StreamId& stream_id,
+    const VersionQuery& version_query
```

Comment:
Yeah we probably should start doing this. Add an extra 4 spaces at the parameter list

Reply:
I agree, but I am not going to change it this time, since all the functions in this file follow the same pattern. I think it is better to do all of them in one go, otherwise it would be worst the change IMO.